snuba/datasets/cdc/cdcprocessors.py

```diff
@@ -15,10 +40,28 @@ def _process_commit(self, offset):
         pass
 
     def _process_insert(self, offset, columnnames, columnvalues):
-        pass
+        return self._message_row_class.from_wal(
+            offset,
+            columnnames,
+            columnvalues
+        ).to_clickhouse()
 
     def _process_update(self, offset, key, columnnames, columnvalues):
-        pass
+        old_key = dict(zip(key['keynames'], key['keyvalues']))
+        new_key = {
+            key: columnvalues[columnnames.index(key)]
+            for key
+            in key['keynames']
+        }
+        # We cannot support a change in the identity of the record
+        # clickhouse will use the identity column to find rows to merge.
+        # if we change it, merging won't work.
+        assert old_key == new_key, 'Changing Primary Key is not supported.'
+        return self._message_row_class.from_wal(
+            offset,
+            columnnames,
+            columnvalues
+        ).to_clickhouse()
 
     def _process_delete(self, offset, key):
         pass
```

Comment:
Why is this different than the other `_process_*` family of messages that are now generalized?

Reply:
There was one but I cannot remember. Let me try to do it again, and then it should come back.

Reply:
oh yes. It was because, while for updates, we can simply create a new GroupedMessageRow with the new value since that value includes the identity key, for deletion this is more complex because we have to create a GroupedMessageRow with only the identity key populated. The rest is None and it is not in the WAL message.
As a result we still need a  method somewhere that knows which fields to populate as identity key (only id here) and what's their name on the GroupedMessageRow class. So I could put that logic in GroupedMessageRow adding a new "forDeletion" method but that does not change much with respect to what we do here.